### ISSUE TITLE
upgrade pyarrow dependency to 0.17

### ISSUE THREAD

**Issue by hc128168** [apache/incubator-superset#9759]
I used to have pyarrow 0.17 (the latest) on my machine.  When I installed the latest superset, it downgraded it to 0.16.

Then, I have to manually upgrade again but it complains that superset 0.36 is incompatible with 0.17:
```
$ pip3.6 install --user --upgrade pyarrow
WARNING: pip is being invoked by an old script wrapper. This will fail in a future version of pip.
Please see https://github.com/pypa/pip/issues/5599 for advice on fixing the underlying issue.
To avoid this problem you can invoke Python with '-m pip' instead of running pip directly.
Collecting pyarrow
  Downloading pyarrow-0.17.0-cp36-cp36m-manylinux2014_x86_64.whl (63.8 MB)
     |████████████████████████████████| 63.8 MB 10.5 MB/s 
Requirement already satisfied, skipping upgrade: numpy>=1.14 in /home/hc/.local/lib/python3.6/site-packages (from pyarrow) (1.18.4)
Could not build wheels for numpy, since package 'wheel' is not installed.
ERROR: apache-superset 0.36.0 has requirement pyarrow<0.17.0,>=0.16.0, but you'll have pyarrow 0.17.0 which is incompatible.

```

There is a bug in an older version of cython that pyarrow 0.16 uses.  Hence, it would be great if superset could change the dependency of pyarrow to 0.17.

Thanks!

**Comment by villebro** [comment]
`0.36.0` has pinned `pyarrow` at `<0.17`, as `0.16.0` was the most recent version available when `0.36.0` was released. This is due to semver not guaranteeing forward compatibility for minor versions. `master` branch of Superset has bumped `pyarrow` to `0.17.0`, hence the next release will automatically pull that in. If you want, you should be able to bump `pyarrow` after you've installed Superset, as in this case there didn't seem to be any breaking changes in the new version that affected Superset, i.e. the error can be ignored.

Note, however, that it is recommended to always run Superset in an isolated virtualenv separately from your system installation to avoid precisely these kinds of problems.

**Comment by hc128168** [comment]
Thanks for clarifying!
